ui-admin/src/forms/designer/split/FullQuestionDesigner.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { useState } from 'react'
+
+import { HtmlQuestion, PortalEnvironmentLanguage, Question, QuestionType } from '@juniper/ui-core'
+
+import { Textarea } from 'components/forms/Textarea'
+import { i18nSurveyText } from 'util/juniperSurveyUtils'
+import { BaseFields } from '../questions/BaseFields'
+import { ChoicesList } from '../questions/ChoicesList'
+import { TextFields } from '../questions/TextFields'
+import { VisibilityFields } from '../questions/VisibilityFields'
+import { TextInput } from 'components/forms/TextInput'
+import { IconButton } from 'components/forms/Button'
+import { faAsterisk, faLock, faUnlock } from '@fortawesome/free-solid-svg-icons'
+import QuestionTypeSelector from './QuestionTypeSelector'
+import classNames from 'classnames'
+import { baseQuestions } from '../questions/questionTypes'
+
+export type QuestionDesignerProps = {
+    question: Question
+    isNewQuestion: boolean
+    readOnly: boolean
+    currentLanguage: PortalEnvironmentLanguage
+    supportedLanguages: PortalEnvironmentLanguage[]
+    onChange: (newValue: Question) => void
+}
+
+/**
+ * UI for editing a question in a form.
+ * Note that this will eventually replace the QuestionDesigner component
+ */
+export const FullQuestionDesigner = (props: QuestionDesignerProps) => {
+  const {
+    question, isNewQuestion, readOnly,
+    onChange, currentLanguage, supportedLanguages
+  } = props
+
+  const isTemplated = 'questionTemplateName' in question
+  const [isStableIdLocked, setIsStableIdLocked] = useState(true)
+
+  return (
+    <div>
+      <label className={'form-label fw-semibold'}>Stable ID</label>
+      <span className="text-danger fw-semibold ms-1">*</span>
+      <div className="d-flex align-items-center">
+        <div className="w-100">
+          <TextInput
+            className={'mb-2'}
+            required={true}
+            value={question.name}
+            disabled={isStableIdLocked}
+            onChange={name => {
+              onChange({
+                ...question,
+                name
+              })
+            }}
+          />
+        </div>
+        <IconButton
+          className="mb-2"
+          icon={isStableIdLocked ? faLock : faUnlock}
+          aria-label={isStableIdLocked ? 'Edit stable ID' : 'Lock stable ID'}
+          onClick={() => setIsStableIdLocked(!isStableIdLocked)}
+        />
+        <IconButton
+          className={classNames('mb-2', { 'text-danger': question.isRequired })}
+          icon={faAsterisk}
+          aria-label={'Toggle required'}
+          onClick={() => {
+            onChange({
+              ...question,
+              isRequired: !question.isRequired
+            })
+          }}
+        />
+      </div>
```

Comment:
Love this! My only confusion is that it's slightly unclear if blue or red means checked - I wonder if we did grey instead of blue it would be clearer? If there was some `optional` icon that would make our lives so much easier, but alas...

Reply:
I just switched this to use text-muted when it's disabled, so there's a little bit more contrast in states. It's still not perfect, but I think we'll probably change this part of the UX once the prototype matures a bit